node/parallel/src/chain_spec/mod.rs

```diff
@@ -15,13 +15,31 @@
 pub mod heiko;
 pub mod parallel;
 
-use primitives::*;
 use sc_chain_spec::{ChainSpecExtension, ChainSpecGroup};
+use sc_service::Properties;
 use serde::{Deserialize, Serialize};
+use serde_json::json;
 use sp_consensus_aura::sr25519::AuthorityId as AuraId;
 use sp_core::{sr25519, Pair, Public};
 use sp_runtime::traits::IdentifyAccount;
 
+use primitives::{network::NetworkType, *};
+
+/// Generate chain properties for network.
+///
+/// For fields definition, see https://github.com/polkadot-js/apps/blob/bd78840d2142df121d182e8700b20308880dde0a/packages/react-api/src/Api.tsx#L115
+pub(crate) fn as_properties(network: NetworkType) -> Properties {
+    let (symbol, decimal) = network.token_info();
+    json!({
+        "ss58Format": network.ss58_addr_format_id(),
+        "tokenSymbol": symbol,
+        "tokenDecimals": decimal,
+    })
+    .as_object()
+    .expect("Network properties are valid; qed")
```

Comment:
```suggestion
    .expect("Network properties are invalid; qed")
```

Reply:
I think it proves why the statement won't be panic.  See https://github.com/paritytech/substrate/blob/1d5abf01abafdb6c15bcd0172f5de09fd87c5fbf/primitives/npos-elections/src/balancing.rs#L122